dowhy/causal_identifier/identify_effect.py

```diff
@@ -0,0 +1,190 @@
+            default_backdoor_id=copy.deepcopy(self.default_backdoor_id),
+            identifier_method=copy.deepcopy(self.identifier_method),
+        )
+
+    def __str__(self, only_target_estimand=False, show_all_backdoor_sets=False):
+        if self.no_directed_path:
+            s = "No directed path from {0} to {1} in the causal graph.".format(
+                self.treatment_variable, self.outcome_variable
+            )
+            s += "\nCausal effect is zero."
+            return s
+        s = "Estimand type: {0}\n".format(self.estimand_type)
+        i = 1
+        has_valid_backdoor = sum("backdoor" in key for key in self.estimands.keys())
+        for k, v in self.estimands.items():
+            if show_all_backdoor_sets:
+                # Do not show backdoor key unless it is the only backdoor set.
+                if k == "backdoor" and has_valid_backdoor > 1:
+                    continue
+            else:
+                # Just show the default backdoor set
+                if k.startswith("backdoor") and k != "backdoor":
+                    continue
+            if only_target_estimand and k != self.identifier_method:
+                continue
+            s += "\n### Estimand : {0}\n".format(i)
+            s += "Estimand name: {0}".format(k)
+            if k == self.default_backdoor_id:
+                s += " (Default)"
+            s += "\n"
+            if v is None:
+                s += "No such variable(s) found!\n"
+            else:
+                sp_expr_str = sp.pretty(v["estimand"], use_unicode=True)
+                s += "Estimand expression:\n{0}\n".format(sp_expr_str)
+                j = 1
+                for ass_name, ass_str in v["assumptions"].items():
+                    s += "Estimand assumption {0}, {1}: {2}\n".format(j, ass_name, ass_str)
+                    j += 1
+            i += 1
+        return s
+
+
+def identify_effect(
+    graph: CausalGraph,
+    treatment: Union[str, List[str]],
+    outcome: Union[str, List[str]],
+    method: CausalIdentifier,
+    node_names=None,
+    conditional_node_names=None,
+):
+    """Identify the causal effect to be estimated based on a CausalGraph
+
+    :param graph: CausalGraph to be analyzed
+    :param treatment: name of the treatment
+    :param outcome: name of the outcome
+    :param method: CausalIdentifier instance to use to identify effects
+    :param node_names: OrderedSet comprising names of all nodes in the graph (Used for IDIdentifier only)
+    :param conditional_node_names: variables that are used to determine treatment. If none are
+    provided, it is assumed that the intervention is static (Used for AutoIdentifier only).
+    :returns: a probability expression (estimand) for the causal effect if identified, else NULL
+    """
+    identified_estimand = method.identify_effect(
+        graph, treatment, outcome, node_names=node_names, conditional_node_names=conditional_node_names
+    )
+    return identified_estimand
```

Comment:
What's the purpose of this function?

Reply:
This function was added, after a conversation with @amit-sharma we decided to add it as the idea is that in the future some logic is going to be added here to help the users set some parameters for them (users that do not know a lot about the topic, don't know what parameters to use, etc). For now, I guess it is just a wrapper to execute the method identify_effect as a function.

Reply:
@amit-sharma Why don't we add this function once we have an actual use case for this additional logic. Right now, this function just adds a layer without benefits. It's even confusing, because it's unclear why you would use this function over just calling the objects' methods directly.

Reply:
Good point, Peter. This is an entry function for beginners, and the idea is that it automatically selects the applicable identification methods and outputs the valid identification strategies. It is useful for users who do not know which method to apply.

But I do see that the implementation is missing :) I think the necessary functionality is already there through AutoIdentifier. @andresmor-ms can you modify this function such that `method` is optional, and if a user does not provide the method then we use the AutoIdentifier with default initialization values (ATE, BACKDOOR_DEFAULT) and output its results? 

@petergtz with this new functionality, would this function make sense?

Reply:
@amit-sharma Yes, sounds good. See also latest comment further below. Resolving this comment.